### PR TITLE
Support flask async

### PR DESCRIPTION
**Pull request by jiashuChen**
Hi, @Fischerfredl,

I'm just trying to add `flask[async]` support, appreciate for any feedback if you are unhappy with the current approach.

I've added new test cases, ensure all unit tests are passing locally.

PS: I've also updated `.gitignore` to exclude jetbrain (pycharm) IDE

```
============================= test session starts ==============================
collecting ... collected 19 items

test_defaults.py::TestDefaults::test_async_default_works PASSED          [  5%]
test_defaults.py::TestDefaults::test_default_gets_validated PASSED       [ 10%]
test_defaults.py::TestDefaults::test_default_off_by_default PASSED       [ 15%]
test_defaults.py::TestDefaults::test_default_works PASSED                [ 21%]
test_defaults.py::TestDefaults::test_setdefault_on_string PASSED         [ 26%]
test_format_validation.py::TestDefaults::test_format_validation_happy_path PASSED [ 31%]
test_format_validation.py::TestDefaults::test_format_validation_off_by_default PASSED [ 36%]
test_format_validation.py::TestDefaults::test_format_validation_rejection PASSED [ 42%]
test_ignore.py::TestExpects::test_default_behaviour PASSED               [ 47%]
test_ignore.py::TestExpects::test_ignore_multiple PASSED                 [ 52%]
test_ignore.py::TestExpects::test_ignore_one PASSED                      [ 57%]
test_requests.py::TestExpects::test_check_mimetype PASSED                [ 63%]
test_requests.py::TestExpects::test_check_null PASSED                    [ 68%]
test_requests.py::TestExpects::test_valid_decorator PASSED               [ 73%]
test_requests.py::TestExpects::test_valid_decorator_no_schema_async PASSED [ 78%]
test_schema.py::TestExpects::test_additional_parameter PASSED            [ 84%]
test_schema.py::TestExpects::test_missing_parameter PASSED               [ 89%]
test_schema.py::TestExpects::test_validation_invalid PASSED              [ 94%]
test_schema.py::TestExpects::test_validation_valid PASSED                [100%]

============================== 19 passed in 0.12s ==============================

Process finished with exit code 0

```

### CODE REVIEW

**Comment by Fischerfredl** [comment]
Hi jiasshuChen, thank you for the Pull Request! Can you give me a link or point to something to read, so I can get familiar with the topic (e.g documentaion/some article)? 

Some questions:
- I see that Flask added async support in version 2.0. Will this library still work with older versions of Flask after the changes?
- Do current async routes not work at all with this library?
- Or will this change enable validation in coroutines, then what would be the benefits?

Can you give a short summary of what the changes are and their benefits?

Thank you
Fischerfredl

**Comment by jiashuChen** [comment]
> Hi jiasshuChen, thank you for the Pull Request! Can you give me a link or point to something to read, so I can get familiar with the topic (e.g documentaion/some article)?
> 
> Some questions:
> 
>     * I see that Flask added async support in version 2.0. Will this library still work with older versions of Flask after the changes?
> 
>     * Do current async routes not work at all with this library?
> 
>     * Or will this change enable validation in coroutines, then what would be the benefits?
> 
> 
> Can you give a short summary of what the changes are and their benefits?
> 
> Thank you Fischerfredl

Hi @Fischerfredl,

You can find an article describing how to add async support to flask extensions here https://flask.palletsprojects.com/en/2.0.x/async-await/#extensions.

1. The code change I am making should ensure the library remain compatible with flask before async support is added.
2. Yes, without my change, the library is not able to support flask async handlers. (You can simply run the unit test in my code change without my change to the library)

File: flask_expects_json/__init__.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ def decorated_function(*args, **kwargs):
                 return abort(400, e)
 
             g.data = data
-            return f(*args, **kwargs)
+
+            if hasattr(current_app, "ensure_sync"):
```
Comment (jiashuChen) [inline flask_expects_json/__init__.py:48]: @Fischerfredl this if statement here should ensure the library is still compatible with old version of flask without async support

**Comment by Fischerfredl** [comment]
Great! The changes look good to me. I will check the tests later today and merge the PR then.

**Pull request merged**